ci: Add anaconda pruning to CI pipeline

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44651 ci: Add anaconda pruning to CI pipeline**

Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23692851](https://our.internmc.facebook.com/intern/diff/D23692851)

seemethere committed:
ci: Add anaconda pruning to CI pipeline

Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add anaconda pruning to CI pipeline"


Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add anaconda pruning to CI pipeline"


Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add anaconda pruning to CI pipeline"


Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

walterddr commented on .circleci/config.yml:
```diff
@@ -1113,6 +1113,35 @@ jobs:
+    parameters:
+      packages:
+        type: string
+        description: "What packages are we pruning? (eg. pytorch, torchvision, torchaudio, etc.)"
```
```suggestion
        description: "What packages are we pruning? (eg. pytorch, torchvision, torchaudio, etc.)"
        description: "What packages are we pruning? (quoted, space-separated string. eg. "pytorch", "torchvision torchaudio", etc.)"
```

walterddr commented on .circleci/verbatim-sources/job-specs/binary-job-specs.yml:
```diff
@@ -324,3 +324,32 @@
+          name: Prune packages
+          no_output_timeout: "1h"
+          command: |
+              ANACONDA_API_TOKEN="${CONDA_PYTORCHBOT_TOKEN}" \
```
this might be a dumb question but where is `CONDA_PYTORCHBOT_TOKEN` set?

seemethere replied:
Not a dumb question.

This is set within circleci's credential store under the `org-member` context.

walterddr reviewed (approved):
lgtm

seemethere committed:
Update on "ci: Add anaconda pruning to CI pipeline"


Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23692851](https://our.internmc.facebook.com/intern/diff/D23692851)

[ghstack-poisoned]

seemethere committed:
Update on "ci: Add anaconda pruning to CI pipeline"


Adds pruning for our anaconda channels (pytorch-nightly, pytorch-test)
into our CI pipeline so that it gets run on a more consistent basis.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23692851](https://our.internmc.facebook.com/intern/diff/D23692851)

[ghstack-poisoned]
